From 69e62417c7d313ebb879eb4a8bd0130149d85949 Mon Sep 17 00:00:00 2001
From: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>
Date: Thu, 14 Jun 2018 23:38:34 +0530
Subject: [PATCH 045/144] arm: dts: s900: Add Clock Management Unit (CMU) nodes

This commit adds Clock Management Unit (CMU) nodes for Actions Semi
S900 SoC.

Signed-off-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>
---
 arch/arm/dts/s900.dtsi | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/arch/arm/dts/s900.dtsi b/arch/arm/dts/s900.dtsi
index 3bd14b82d4..e9d47b1ff1 100644
--- a/arch/arm/dts/s900.dtsi
+++ b/arch/arm/dts/s900.dtsi
@@ -6,18 +6,40 @@
 // Copyright (C) 2018 Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>
 
 /dts-v1/;
+#include <dt-bindings/clock/s900_cmu.h>
 
 / {
 	compatible = "actions,s900";
 	#address-cells = <0x2>;
 	#size-cells = <0x2>;
 
+	losc: losc {
+		compatible = "fixed-clock";
+		clock-frequency = <32768>;
+		#clock-cells = <0>;
+	};
+
+	diff24M: diff24M {
+		compatible = "fixed-clock";
+		clock-frequency = <24000000>;
+		#clock-cells = <0>;
+	};
+
 	soc {
 		u-boot,dm-pre-reloc;
 		compatible = "simple-bus";
 		#address-cells = <0x2>;
 		#size-cells = <0x2>;
 		ranges;
+
+		cmu: clock-controller@e0160000 {
+			u-boot,dm-pre-reloc;
+			compatible = "actions,s900-cmu";
+			reg = <0x0 0xe0160000 0x0 0x1000>;
+			clocks = <&losc>, <&diff24M>;
+			clock-names = "losc", "diff24M";
+			#clock-cells = <1>;
+		};
 	};
 };
 
-- 
2.18.0

